classes/ActionScheduler_wpPostStore.php

```diff
@@ -11,7 +11,7 @@ class ActionScheduler_wpPostStore extends ActionScheduler_Store {
 	/** @var DateTimeZone */
 	protected $local_timezone = NULL;
 
-	public function save_action( ActionScheduler_Action $action, DateTime $date = NULL ){
+	public function save_action( ActionScheduler_AbstractAction $action, DateTime $date = NULL ){
```

Comment:
If possible it would be better to type hint for an interface rather than a class, even an abstract class. With that said, that may not be as feasible without a bunch of rewriting, so if it won't be easy then we can keep it as is.

Also, `null` should be lowercase.

Reply:
> With that said, that may not be as feasible without a bunch of rewriting

I don't think it would require any rewriting if we used `ActionScheduler_Interface_Action`. It would still maintain backward compatibility as `ActionScheduler_AbstractAction` implements it, and `ActionScheduler_Action` extends `ActionScheduler_AbstractAction`.

Can you share a good article or two explaining why it's better to help me learn?